src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2116 @@
+                        # No event (yet) => fine
+                        pass
+
+                rargs = self.parent.queue_img.get(block=True)  # block until receive something
+                logging.debug("Received message %s", rargs)
+
+                if rargs is None:  # if message is None end the thread
+                    return
+
+                # synchronized mode
+                if self._sync_event:
+                    if rargs == "start":
+                        logging.info("Received event trigger")
+                        continue
+                    else:
+                        logging.info("Get the synchronized image.")
+
+                # non-sync mode
+                else:
+                    while not self.parent.queue_img.empty():
+                        # keep reading to check if there might be a newer image for display
+                        # in case we are too slow with reading
+                        rargs = self.parent.queue_img.get(block=False)
+
+                        if rargs is None:  # if message is None end the thread
+                            return
+                    logging.info("No more images in queue, so get the image.")
+
+                if rargs == "F":  # Flush => the previous images are from the previous acquisition
+                    logging.debug("Acquisiton was stopped so flush previous images.")
+                    continue
+
+                self._metadata[model.MD_ACQ_DATE] = time.time() - \
+                                                    (self.exposureTime.value + self._metadata[model.MD_READOUT_TIME])
+
+                # get the image from the buffer
+                img_num = rargs[1]
+                img_info = self.parent.ImgRingBufferGet("Data", img_num)
+
+                if not img_info:  # TODO check if this ever happens in log and if not remove!
+                    logging.warning("Image info received from buffer is empty!")
+                    continue
+
+                img_size = int(img_info[0]) * int(img_info[1]) * 2  # num of bytes we need to receive #TODO why 2?
+                img_num_actual = img_info[4]
+
+                img = ""
+                try:
+                    while len(img) < img_size:  # wait until all bytes are received
+                        img += self.parent._dataport.recv(img_size)
+                except socket.timeout as msg:
+                    logging.error("Did not receive an image: %s", msg)
+                    continue
+
+                image = numpy.frombuffer(img, dtype=numpy.uint16)  # convert to array
+                image.shape = (int(img_info[1]), int(img_info[0]))
+
+                logging.debug("Requested image number %s, received image number %s from buffer."
+                              % (img_num, img_num_actual))
+
+                # get the scaling table to correct the time axis
+                # TODO only request scaling table if corresponding MD not available for this time range
```

Comment:
I'd suggest to add a log message just before and just after this code block to check how long it takes to receive the metadata. If it's always very short, we could leave this simple code as-is.

Reply:
message should include the time, or will we use the time for the logging event